File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxprofit = 0;
     let min = prices[0];
    let max;
    for(let i=0; i<prices.length; i++){
        if(prices[i]<min){
            min = prices[i];
        }
        max = prices[i]-min; 
        if(maxprofit<max){
        maxprofit = max;
         }
    }
    return maxprofit;
    
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int currentprofit = 0;
        int maxprofit = 0;

        for(int i=1; i<prices.length; i++){
            int diff = prices[i]-prices[i-1];
            currentprofit = currentprofit+diff;

            if(currentprofit<0){
                currentprofit = 0;
            }
        
           if(maxprofit<currentprofit){
            maxprofit = currentprofit;
           }

        }
        return maxprofit;
        
    }
}